bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueBridgeHandler.java

```diff
@@ -774,7 +775,11 @@ private void notifySensorStatusListeners(final FullSensor fullSensor, final Stri
      * @return {@code true} if the available information of both states are equal.
      */
     private boolean isEqual(State state1, State state2) {
-        return state1.getAlertMode().equals(state2.getAlertMode()) && state1.isOn() == state2.isOn()
+        AlertMode alertMode1 = state1.getAlertMode();
+        AlertMode alertMode2 = state2.getAlertMode();
+
+        return ((alertMode1 == null && alertMode2 == null) || (alertMode1 != null && alertMode1.equals(alertMode2)))
```

Comment:
```suggestion
        return Objects.equals(state1.getAlertMode(), state2.getAlertMode())
```

Reply:
This is a good point. I already thought about this and in my opinion it would be more Java-Standardconform to implement an equals method inside the state class and replace the calls of this isEqual(state1, state2) by state1.equals(state2)

Reply:
I have refactored the State class by adding the equals method. There is one change to previous isEqual method at HueBridgeHandler used before. The attribute float[] xy of State Class was not compared in isEqual() method and is now compared by equals method. As I think there is no affect because of this logic change, but perhaps someone of you can verify this.